spec/controllers/admin/movies_controller_spec.rb

```diff
@@ -0,0 +1,167 @@
+require "rails_helper"
+include SessionsHelper
+RSpec.describe Admin::MoviesController, type: :controller do
+  let(:category1){FactoryBot.create :category}
+  let(:category2){FactoryBot.create :category}
+  let(:movie1){FactoryBot.create :movie}
+  let(:movie2){FactoryBot.create :movie}
+  let!(:cinema){FactoryBot.create :cinema}
+  let!(:room){FactoryBot.create :room, cinema_id: cinema.id}
+  let!(:show_time) do
+    FactoryBot.create :show_time, start_time: (Date.today + 1).to_s + " 10:15:15.000000000 UTC +00:00", 
+                                movie_id: movie1.id, 
+                                room_id: room.id
+  end
+  let(:user){FactoryBot.create :user}
+  before do 
+    log_in user
+  end
+  describe "GET#index" do
+    it "should show the list of movies" do
+      get :index
+      expect(assigns(:movies)).to eq([movie1, movie2])
+    end
+  end 
+  describe "GET#show" do 
+    context "failed when movie not found" do 
+      it "flash error film ot found" do 
```

Comment:
context nên bắt đầu bằng when nha, it thì bắt đầu bằng should, dịch sang tiếng việt câu này là kiểu miêu tả Get#show, "khi" mà nó thế này thì nó "nên" thế này